pulsar-client/src/main/java/org/apache/pulsar/client/impl/BatchMessageContainerImpl.java

```diff
@@ -369,6 +372,9 @@ public boolean hasSameSchema(MessageImpl<?> msg) {
         if (!messageMetadata.hasSchemaVersion()) {
             return msg.getSchemaVersion() == null;
         }
+        if (messageMetadata.hasSchemaId()) {
+            return Arrays.equals(msg.getSchemaId(), messageMetadata.getSchemaId());
+        }
```

Comment:
When users use the external scheam, what the schema version would be? 
We should still check the scheam id equality if the scheam version is set to null.